src/widget/container.rs

```diff
@@ -0,0 +1,67 @@
+use super::Widget;
+use embedded_graphics::{
+    prelude::*,
+    primitives::{CornerRadii, PrimitiveStyleBuilder, Rectangle, RoundedRectangle},
+};
+
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Border<Color> {
```

Comment:
Different kinds of displays will have different color formats (B&W vs RGB, different bit depths, etc.), so everything needs to be parameterized by the color being used.